variables.tf

```diff
@@ -93,6 +93,109 @@ variable "api_server_authorized_ip_ranges" {
   default     = null
 }
 
+variable "auto_scaler_profile_balance_similar_node_groups" {
+  description = "Enable or Disable the balance similar node groups. Defaults to false."
+  type        = bool
+  default     = false
+}
+
+variable "auto_scaler_profile_empty_bulk_delete_max" {
+  description = "The maximum number of empty nodes that can be deleted at the same time. Defaults to 10."
+  type        = number
+  default     = 10
+}
+
+variable "auto_scaler_profile_expander" {
+  description = "The expander for the node group. Possible values are `random`, `most-pods`, `least-waste`, `price`, `priority`. Defaults to `random`."
+  type        = string
+  default     = "random"
+}
+
+variable "auto_scaler_profile_max_graceful_termination_sec" {
+  description = "The maximum time in seconds for which a node can be gracefully terminated. Defaults to 600."
+  type        = number
```

Comment:
It looks like the type should be [string](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L136-L140) here.